src/Http/Routing/src/Builder/RouteHandlerFilterExtensions.cs

```diff
@@ -41,9 +47,21 @@ public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, IR
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
     /// <param name="routeHandlerFilter">A <see cref="Delegate"/> representing the core logic of the filter.</param>
     /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
-    public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, Func<RouteHandlerFilterContext, Func<RouteHandlerFilterContext, ValueTask<object?>>, ValueTask<object?>> routeHandlerFilter)
+    public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, Func<RouteHandlerInvocationContext, RouteHandlerFilterDelegate, ValueTask<object?>> routeHandlerFilter)
     {
-        builder.RouteHandlerFilters.Add(new DelegateRouteHandlerFilter(routeHandlerFilter));
+        builder.RouteHandlerFilterFactories.Add((methodInfo, next) => (context) => routeHandlerFilter(context, next));
```

Comment:
```suggestion
        builder.RouteHandlerFilterFactories.Add((routeHandlerContext, next) => (context) => routeHandlerFilter(context, next));
```

Reply:
*faceplam* These rebases will be the death of me.